go/client/cmd_wallet_send.go

```diff
@@ -72,6 +82,22 @@ func (c *CmdWalletSend) ParseArgv(ctx *cli.Context) error {
 		}
 	}
 	c.Note = ctx.String("message")
+
+	c.Memo = ctx.String("memo")
+	memoType := ctx.String("memo_type")
+	if memoType == "" {
+		if c.Memo != "" {
+			memoType = "text"
+		} else {
+			memoType = "none"
+		}
+	}
+	var ok bool
+	c.MemoType, ok = stellar1.PublicNoteTypeMap[strings.ToUpper(memoType)]
```

Comment:
it's generally unfortunate that we have integers, lowercase strings, and uppercase strings which are all magic values corresponding to memo types. :/